### ISSUE TITLE
Modern test helpers

### ISSUE THREAD

**Issue by jgwhite** [adopted-ember-addons/ember-sortable#256]
It would be great to add `drag` and `reorder` test helpers using the modern ember test helper patterns. The outcome is that apps using “classic” (globals) test helpers can continue as-is whilst modern apps can import helpers as follows:

```js

import { module, test } from 'qunit';
import { visit, currentURL } from '@ember/test-helpers';
import { reorder } from 'ember-sortable/test-helpers';
import { setupApplicationTest } from 'ember-qunit';

module('Acceptance | hello', function(hooks) {
  setupApplicationTest(hooks);

  test('visiting /hello', async function(assert) {
    await visit('/hello');
    await reorder(
      'mouse',
      'li',
      '[data-test-id="3"]',
      '[data-test-id="2"]',
      '[data-test-id="1"]'
    );

    assert.equal(currentURL(), '/hello');
  });
});

**Comment by ygongdev** [comment]
Here's one for `drag` using the modern test helpers and vanilla JS. However, if we want to keep the `registerAsyncHelper` for backward compatibility, we might have to rename the modern test helper, else there might be a name collision.

```javascript
import { triggerEvent, settled, find } from '@ember/test-helpers';
/**
 * Vanilla Javascript equivalent of JQuery's `.offset`.
 * Reference: https://github.com/nefe/You-Dont-Need-jQuery
 *
 * @method getOffset
 * @param {Element} el an element
 */
function getOffset(el) {
  const box = el.getBoundingClientRect();

  return {
    top: box.top + window.pageYOffset - document.documentElement.clientTop,
    left: box.left + window.pageXOffset - document.documentElement.clientLeft,
  };
}

/**
 * A re-implementation of the drag test helper that is compatible with the new qunit syntax without `registerAsyncHelper`.
 *
 * Drags elements by an offset specified in pixels.
 * Examples
 *    drag(
 *      'mouse',
 *      '.some-list li[data-item=uno]',
 *      function() {
 *        return { dy: 50, dx: 20 };
 *      }
 *    );
 * @method sortableDrag
 * @param {'mouse'|'touch'} [mode]
 *   event mode
 * @param {String} [itemSelector]
 *   selector for the element to drag
 * @param {Function} [offsetFn]
 *   function returning the offset by which to drag
 * @param {Object} [callbacks]
 *   callbacks that are fired at the different stages of the interaction
 * @return {Promise}
 */

export default async function sortableDrag(
  mode,
  itemSelector,
  offsetFn,
  callbacks = {}
) {
  let start;
  let move;
  let end;
  let which;

  if (mode === 'mouse') {
    start = 'mousedown';
    move = 'mousemove';
    end = 'mouseup';
    which = 1;
  } else if (mode === 'touch') {
    start = 'touchstart';
    move = 'touchmove';
    end = 'touchend';
  } else {
    throw new Error(`Unsupported mode: '${mode}'`);
  }

  await settled();

  const itemElement = find(itemSelector);
  const itemOffset = getOffset(itemElement);
  const offset = offsetFn();
  const rect = itemElement.getBoundingClientRect();
  // firefox gives some elements, like <svg>, a clientHeight of 0.
  // we can try to grab it off the parent instead to have a better
  // guess at what the scale is.
  // https://bugzilla.mozilla.org/show_bug.cgi?id=874811#c14
  // https://stackoverflow.com/a/13647345
  // https://stackoverflow.com/a/5042051
  const dx = offset.dx || 0;
  const dy = offset.dy || 0;
  const clientHeight =
    itemElement.clientHeight ||
    itemElement.offsetHeight ||
    itemElement.parentNode.offsetHeight;
  const scale = clientHeight / (rect.bottom - rect.top);
  const halfwayX = itemOffset.left + (dx * scale) / 2;
  const halfwayY = itemOffset.top + (dy * scale) / 2;
  const targetX = itemOffset.left + dx * scale;
  const targetY = itemOffset.top + dy * scale;

  await triggerEvent(itemElement, start, {
    clientX: itemOffset.left,
    clientY: itemOffset.top,
    which,
  });

  if (callbacks.dragstart) {
    await callbacks.dragstart();
  }

  await triggerEvent(itemElement, move, {
    clientX: itemOffset.left,
    clientY: itemOffset.top,
  });

  if (callbacks.dragmove) {
    await callbacks.dragmove();
  }

  await triggerEvent(itemElement, move, {
    clientX: halfwayX,
    clientY: halfwayY,
  });

  await triggerEvent(itemElement, move, {
    clientX: targetX,
    clientY: targetY,
  });

  await triggerEvent(itemElement, end, {
    clientX: targetX,
    clientY: targetY,
  });

  if (callbacks.dragend) {
    await callbacks.dragend();
  }
}
```

**Comment by jgwhite** [comment]
@ygongdev lovely stuff, thanks!

I think it’s OK for `import { drag } from 'ember-sortable/test-helpers';` to clobber the global `registerAsyncHelper` version.

Could you wrap this up in a PR?

**Comment by ygongdev** [comment]
Yeah I can take a stab at it when I have some time.

**Comment by ygongdev** [comment]
I tried to convert `drag` and `reorder` to the new syntax, migrate existing tests to new test syntax, and remove dependency on jquery. However, it doesn't seem like the sorting is ever happening in the tests, thus all of them failed, so I'll need to keep into it more.

**Comment by jgwhite** [comment]
@ygongdev we’ll need to keep the tests for the classic-style helpers around too, to maintain backwards compatibility.

**Comment by nlfurniss** [comment]
Modern test helper should also be moved to `addon-test-support` and out of the `helpers` dir since those are bundled with the addon and into the consuming app :-(

**Comment by kdamball** [comment]
@ygongdev I managed to find the bug related to [your comment above](https://github.com/heroku/ember-sortable/issues/256#issuecomment-498469701). When combined with [this issue](https://github.com/heroku/ember-sortable/issues/234), or rather [it's fix](https://github.com/heroku/ember-sortable/issues/234#issuecomment-498589111) - it sorts as expected.

**TL;DR**: add to you Sass files 
```scss
.sortable-item {
  transition: all .125s;
  
  &.is-dragging {
    transition-duration: 0s;
  }
}
```

**Comment by ygongdev** [comment]
Change merged in #271
